src/seedu/addressbook/data/person/Block.java

```diff
@@ -8,8 +10,11 @@
 public class Block {
     private String block;
 
-    public Block(String block) {
-        this.block = block;
+    public static String REGEX = "(^[\\d]{1,4}[A-Z]{0,1}$)|(^$)";
```

Comment:
(^[\\d]{1,4}[A-Z]{0,1}$) - begin with 1 to 4 digits followed by optional capital letter
^$ - empty
Block can be empty because not all addresses have block numbers
I did throw a message in the constructor if the regex matching fails which documents this but a better way is to have a ERROR_MESSAGE class variable on top of REGEX